cilium/CFP-36975-configuration-profiles.md

```diff
@@ -0,0 +1,121 @@
+# Cilium Feature Proposal: Configuration Profiles
+
+## Author(s)
+
+Dorde Lapcevic (dordel@google.com)
+
+## Date
+
+2025-02-07
+
+## Status
+
+Implementable
+
+## Abstract
+
+This proposal introduces the concept of "Configuration Profiles" to Cilium.  Profiles are pre-defined, documented, and tested sets of Cilium configuration options designed to address specific operational needs and use cases.  This aims to simplify deployment, improve predictability, and enhance the user experience by providing well-defined operational modes.
+
+## Motivation
+
+Currently, Cilium offers numerous feature flags that operators configure individually.  This granular control, while powerful, presents several challenges:
+
+*   **Combinatorial Complexity:** The interaction between numerous flags can lead to unexpected behavior and makes exhaustive testing difficult.  Certain combinations may be subtly broken or have performance implications that are not immediately apparent.
+*   **Operational Expertise Required:**  Operators need a deep understanding of individual flags and their interplay to configure Cilium optimally for their environment.
+*   **Lack of Clear Guidance:**  New users may struggle to understand the best configuration for their needs, leading to suboptimal deployments.
+*   **Implicit Knowledge:**  Experienced operators often develop a set of preferred flags based on their specific environment (scale, performance requirements, network topology, etc.). This knowledge is often implicit and not easily shared.
+
+Configuration Profiles address these issues by:
+
+*   **Simplifying Deployment:**  Profiles provide a "one-click" (or "one-command") deployment option for common use cases.
+*   **Improving Predictability:**  Each profile will be thoroughly tested and documented, ensuring consistent behavior and performance.
+*   **Providing Clear Guidance:** Documentation will clearly explain the purpose, benefits, limitations, and enabled/disabled features of each profile.
+*   **Best Practices:** Profiles encapsulate expert knowledge and best practices for specific operational scenarios.
+*   **Targeted Testing:** By testing at the profile level (a set of features) instead of only individual flags, we increase coverage and reduce the risk of unexpected interactions.
+
+## Goals
+
+The main goals of this proposal are:
+*   Define a process for creating a new Configuration Profile
+*   Establish a testing framework to ensure the stability and correctness of each profile.
+*   Enable the community to propose and contribute new profiles.
```

Comment:
defining what features works with this profile

Reply:
Added.